src/AuthenticatedAPIClient/axiosConfig.js

```diff
@@ -78,10 +78,11 @@ function applyAxiosInterceptors(authenticatedAPIClient) {
           queueRequests = false;
           PubSub.publishSync(ACCESS_TOKEN_REFRESH);
         })
-        .catch(() => {
-          // TODO: We should give the client app an opportunity to
-          // take control here before logout/redirect to sign in.
-          authenticatedAPIClient.logout();
+        .catch((error) => {
+          if (!authenticatedAPIClient.handleRefreshAccessTokenFailure
+            || authenticatedAPIClient.handleRefreshAccessTokenFailure(error)) {
+            authenticatedAPIClient.logout();
```

Comment:
Also, I'm not sure if we want `logInfo` or not in case we have too many, but could we add a comment like:
```
// This is likely an expected situation where the user really is logged out, but we use logout to ensure clean-up before automatically being redirected to login
```
@douglashall: Is this accurate? ^

Reply:
Yes, this comment is accurate.